conductr_cli/conductr_restore.py

```diff
@@ -0,0 +1,114 @@
+import json
+import os
+import shutil
+import tempfile
+
+import io
+import copy
+import logging
+
+from conductr_cli import control_protocol, bundle_utils, validation, conduct_load, bundle_scale
+from conductr_cli.bundle_core_info import BundleCoreInfo
+
+
+@validation.handle_connection_error
+@validation.handle_http_error
+@validation.handle_wait_timeout_error
+def restore(args):
+    log = logging.getLogger(__name__)
+    destination_bundles = control_protocol.get_bundles(args)
+    destination_bundles_info = BundleCoreInfo.from_bundles(destination_bundles)
+
+    restore_directory = unpack_backup(args.backup)
+    bundle_json_file = open(os.path.join(restore_directory, 'bundles.json'), 'r')
+    bundles_json = bundle_json_file.read()
+    bundle_json_file.close()
+
+    bundles_info = sorted(BundleCoreInfo.from_bundles(json.loads(bundles_json)), key=lambda b: b.start_time)
+
+    restore_errors = []
+    for bundle_info in bundles_info:
+        new_bundle_id = None
+        log.info('\nRestoring bundle : {}.'.format(bundle_info.bundle_name))
+
+        try:
+            new_bundle_id = process_bundle(args, restore_directory, bundle_info)
+            log.info('Loaded {} with bundleId : {}'.format(bundle_info.bundle_name, new_bundle_id))
+        except:
+            restore_errors.append('{} could not be loaded.'.format(bundle_info.bundle_name))
+
+        try:
+            if new_bundle_id is not None:
+                affinity = compatible_bundle(destination_bundles_info, bundle_info.bundle_name,
+                                             bundle_info.compatibility_Version)
+                if affinity == new_bundle_id:
+                    affinity = None
+
+                scale_bundle(args, new_bundle_id, bundle_info.scale, affinity)
+                log.info('Scaled {} to : {}.'.format(new_bundle_id, bundle_info.scale))
+        except:
+            restore_errors.append('{} could not be scaled.'.format(bundle_info.bundle_name))
+
+    for error in restore_errors:
+        log.error(error)
+
```

Comment:
We need to return a boolean here so the exit code is 0 if there are no errors, perhaps: `return len(restore_errors) == 0`. Similarly, this needs to be done for `backup` too which currently exits with 1 when a backup is successful.